### PR TITLE
Improving eBPF support checks

### PR DESCRIPTION
**Pull request by amitschendel**

### CODE REVIEW

File: internal/validator/ebpf/verifier.go [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+	cmd := exec.Command("sh", "-c", "grep -q CONFIG_DEBUG_INFO_BTF=y /boot/config-$(uname -r)")
+	if err := cmd.Run(); err == nil {
+		return nil // BTF is enabled in kernel config
+	}
+
+	return errors.New("BTF support not detected")
+}
+
+// CheckEBPFSupport checks for eBPF support.
+func checkEBPFSupport() error {
```
Comment (matthyx) [inline internal/validator/ebpf/verifier.go:57]: maybe rename to checkKernelVersion?

**Review by matthyx — DISMISSED** [review]
so instead of checking <5.4 we check <4.4 with BTF?

**Comment by amitschendel** [comment]
> so instead of checking <5.4 we check <4.4 with BTF?

We are checking for BTF which is mandatory for our sensor and ig.
Regarding the kernel version, checking 5.4 is because since 5.4 there is BTF/CO-RE but some kernels "lies" with their version so it's not valuable to check it. 
Since kernel 4.4 there is eBPF so we can check that just in case but again, it's not too much valuable but it's more for us to have more debugging information, this is why we don't exist on the error but log it.

What do you think? @matthyx

**Comment by matthyx** [comment]
Fine go ahead

**Pull request merged**